net/yggdrasil/files/yggdrasil.sh

```diff
@@ -16,85 +16,69 @@ proto_yggdrasil_setup_peer_if_non_interface() {
-	if [ -z ${peer_interface} ]; then
-		json_add_string "" ${peer_address}
-	fi;
+	config_get peer_address "$peer_config" "address"
+	config_get peer_interface "$peer_config" "interface"
+	[ -z "$peer_interface" ] && json_add_string "" "$peer_address"
 }
 
 proto_yggdrasil_dump_peer_interface() {
 	local peer_config="$1"
 	local peer_interface
 
-	config_get peer_interface "${peer_config}" "interface"
+	config_get peer_interface "$peer_config" "interface"
 
-	if [ ! -z ${peer_interface} ]; then
-		peer_interfaces="${peer_interfaces}\n${peer_interface}"
-	fi;
+	peer_interfaces="${peer_interfaces}${peer_interface:+\n$peer_interface}"
 }
 
 proto_yggdrasil_setup_peer_if_interface() {
 	local peer_config="$1"
 	local peer_address
 	local peer_interface
-	config_get peer_interface "${peer_config}" "interface"
-	if [ "${peer_interface}" = "${peer_interface_filter}" ]; then
-		config_get peer_address "${peer_config}" "address"
-		json_add_string "" ${peer_address}
+	config_get peer_interface "$peer_config" "interface"
+	if [ "$peer_interface" = "$peer_interface_filter" ]; then
+		config_get peer_address "$peer_config" "address"
+		json_add_string "" "$peer_address"
 	fi;
 }
 
 proto_yggdrasil_append_to_interface_regex() {
-	if [ -z "${regex}" ]; then
-		regex="$1"
-	else
-		regex="${regex}|$1";
-	fi;
+	regex="${regex:+$regex|}$1";
 }
 
 proto_yggdrasil_setup_multicast_interface() {
 	local interface_config="$1"
 	local beacon
 	local listen
-	local port=0
+	local port
 	local password
-	local regex=""
+	local regex=
 
-	config_get beacon "${interface_config}" "beacon"
-	config_get listen "${interface_config}" "listen"
-	config_get port "${interface_config}" "port"
-	config_get password "${interface_config}" "password"
+	config_get beacon   "$interface_config" "beacon"
+	config_get listen   "$interface_config" "listen"
+	config_get password "$interface_config" "password"
+	config_get port     "$interface_config" "port" "0"
 
 	json_add_object ""
-	json_add_boolean "Beacon" $beacon
-	json_add_boolean "Listen" $listen
-	if [ ! -z ${port} ]; then
-		json_add_int "Port" $port
-	else
-		json_add_int "Port" 0
-	fi;
-	if [ ! -z ${password} ]; then
-		json_add_string "Password" $password
-	fi;
-
-	config_list_foreach "${interface_config}" interface proto_yggdrasil_append_to_interface_regex
-
+	json_add_boolean "Beacon" "$beacon"
+	json_add_boolean "Listen" "$listen"
+	json_add_int "Port" "$port"
+	[ -n "$password" ] && json_add_string "Password" "$password"
+	config_list_foreach "$interface_config" interface proto_yggdrasil_append_to_interface_regex
 	json_add_string "Regex" "^(${regex})\$"
-
 	json_close_object
 }
 
 proto_yggdrasil_add_string() {
-	json_add_string "" $1
+	json_add_string "" "$1"
 }
 
 proto_yggdrasil_generate_keypair() {
+	local PrivateKey
 	json_load "$(yggdrasil -genconf -json)"
 	json_get_vars PrivateKey
 	json_cleanup
-	private_key=$PrivateKey
-	public_key=${PrivateKey:64}
+	private_key="$PrivateKey"
+	public_key="${PrivateKey:64}"
```

Comment:
the end to end testing worked OK
the output from key-generation binary will always be a 128 character alpha-numerical without spaces. i guess we can take this, i just tested this seems the same.

Reply:
I'd guess that the yggdrasil binary itself shortens the public key, because like I said ash does not modify the variable on that command:
```sh
xg-135r3 in ~ # PrivateKey=sdfnkoweqhio3iyeroiqhfqewojcpowqjioheqrw
xg-135r3 in ~ # echo "${PrivateKey:2}"
fnkoweqhio3iyeroiqhfqewojcpowqjioheqrw
```

So you may want to change it to "${PrivateKey:0:64}" or "${PrivateKey:0:63}".

Reply:
yggdrasil does not provide the public key when using `-json` output. so, its up to us to create it.

Reply:
I'm not arguing creation of the public key, I'm saying that in ash, `"${PrivateKey:64}"` is the same as `"$PrivateKey"`.